### PR TITLE
add wgARENamespaces to ManageWikiNamespaces

### PR DESCRIPTION
**Pull request by The-Voidwalker**
See https://meta.miraheze.org/w/index.php?title=Community_noticeboard&oldid=156690#Problem_with_ArticleRatings_extension for extra details

### CODE REVIEW

**Comment by dmehus** [comment]
This LGTM, as far as I can tell. Are there any custom configurations for this extension-added variable on existing wikis we'll need to migrate to ManageWiki/namespaces as well?

**Comment by The-Voidwalker** [comment]
To my knowledge this variable has never been configured before, which is honestly impressive, considering it appears to be required to use the extension. 🙃

**Comment by dmehus** [comment]
> To my knowledge this variable has never been configured before, which is honestly impressive, considering it appears to be required to use the extension. 🙃

Wow, heh. 👍

**Review by Universal-Omega — CHANGES_REQUESTED** [review]
This needs to be added to LocalSettings.php also.

File: ManageWikiNamespaces.php [hunk 1]
```diff
@@ -141,4 +141,14 @@
...
 		'requires' => [],
 	],
+	'wgARENamespaces' => [
+		'name' => 'Enable Article Ratings in this namespace',
+		'from' => 'articleratings',
+		'type' => 'check',
+		'main' => true,
+		'talk' => false,
+		'blacklisted' => [],
+		'overridedefault' => false,
```
Comment (Universal-Omega) [inline ManageWikiNamespaces.php:151]: Should be default to be on on $wgContentNamespaces (https://github.com/wikimedia/mediawiki-extensions-ArticleRatings/blob/5ca30436ade43d27edc9cb2669d2ab8395d06734/ArticleRatingsHooks.php#L10)

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

**Comment by The-Voidwalker** [comment]
> This needs to be added to LocalSettings.php also.

I had looked into that, and it seems that none of the other settings in this file are in LS, so I just decided to leave it. Also, to my knowledge any setting in LS.php would be overwritten by the wiki config anyway.

As for setting the default, I'm not actually sure if `overridedefault` (in this context) works the way we'd be expecting it to. I'll take a look.

File: ManageWikiNamespaces.php [hunk 2]
```diff
@@ -141,4 +141,14 @@
...
 		'requires' => [],
 	],
+	'wgARENamespaces' => [
+		'name' => 'Enable Article Ratings in this namespace',
+		'from' => 'articleratings',
+		'type' => 'check',
+		'main' => true,
+		'talk' => false,
+		'blacklisted' => [],
+		'overridedefault' => array_fill_keys( $wgContentNamespaces, true ),
```
Comment (Universal-Omega) [inline ManageWikiNamespaces.php:151]: Should be overridedefault => default => format I believe

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

**Review by Universal-Omega — APPROVED** [review]

File: ManageWikiNamespaces.php (hunk 2 above)
Comment (Universal-Omega) [inline ManageWikiNamespaces.php:151]: will need to have the `'default' => false,` value also

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

**Review by Universal-Omega — APPROVED** [review]

**Pull request merged**